src/manybody_gal.F

```diff
@@ -312,18 +335,9 @@ FUNCTION angular(gal, r_last_update_pbc, iparticle, cell, particle_set, nvec, en
       ENDDO
 
       ! Abort if the oxygen is not part of a water molecule (2 H)
-      IF (count_h == 0) THEN
-         WRITE (*, *) "Faulty atom :", iparticle
-         CPABORT("No H atoms for O found")
-      ELSEIF (count_h == 1) THEN
-         WRITE (*, *) "Faulty atom :", iparticle
-         CPABORT("Only one H atom of O atom found")
-      ELSEIF (count_h == 3) THEN
-         WRITE (*, *) "Faulty atom :", iparticle
-         CPABORT("Three H atoms for O atom found")
-      ELSEIF (count_h > 3) THEN
-         WRITE (*, *) "Faulty atom :", iparticle
-         CPABORT("Error in gal part: too many H atoms for O")
+      IF (count_h /= 2) THEN
+         CALL cp_abort(__LOCATION__, &
+                       " Error: Found "//cp_to_string(count_h)//" H atoms for O atom "//cp_to_string(iparticle))
       ENDIF
 
       IF (count_h == 2) THEN
```

Comment:
This `IF` statement is not needed because you are already aborting in case `count_h /= 2`. The same goes for the occurrence below.

Reply:
Indeed, I will change that.

Reply:
Thanks.